hypothesis-python/src/hypothesis/stateful.py

```diff
@@ -800,14 +806,15 @@ def rule(
     def accept(f):
         if getattr(f, INVARIANT_MARKER, None):
             raise InvalidDefinition(
-                "A function cannot be used for both a rule and an invariant.",
-                Settings.default,
+                "A function cannot be used for both a rule and an invariant. "
+                f"Function: {get_pretty_function_description(f)}",
             )
         existing_rule = getattr(f, RULE_MARKER, None)
         existing_initialize_rule = getattr(f, INITIALIZE_RULE_MARKER, None)
         if existing_rule is not None or existing_initialize_rule is not None:
             raise InvalidDefinition(
-                "A function cannot be used for two distinct rules. ", Settings.default
+                "A function cannot be used for two distinct rules. "
+                f"Function: {get_pretty_function_description(f)}",
             )
         preconditions = getattr(f, PRECONDITIONS_MARKER, ())
```

Comment:
I'm not sure why these errors include `settings.default`, which doesn't seem relevant to tracking down the error. I blamed to https://github.com/HypothesisWorks/hypothesis/pull/2882/files, maybe you have some idea?

Reply:
Seems fine to just cut it.